### DISCUSSION TITLE
Custom Shortcuts

### DISCUSSION THREAD

**Discussion by protean-onion** [TriliumNext/Trilium#2202] [Q&A] [answered]
Hello,

This discussion is regarding custom keyboard shortcuts.

I am not an experienced JavaScript programmer, so I do not understand the structure of the application and, therefore, am unable to customize it for my needs. I find myself inserting equations and highlighting text often for my math course. Having to interrupt typing, right-click, hover over the equation insertion button, and then click it repeatedly seems quite tedious.

It would be great if we had the option to create shortcuts for such tasks, so that we could insert equations, for example, with a convenient keyboard combination. Will writing a script work, or will I have to implement it in some other way? Could somebody please help me with this?

Thanks for your patience.

**Comment by abitofevrything** [comment, marked as answer]
There are already shortcuts available in the options tab if the settings.

By default, the shortcut for "insert math" is Ctrl+M.

**Reply by protean-onion** [reply]
There is no such keyboard shortcut in my version of Trilium. I have ensured that I have the most recent version installed. I have attached a screenshot of the _Keyboard shortcuts_ tab with "math" entered in the search field.
![Screenshot from 2021-10-06 12-07-38](https://user-images.githubusercontent.com/47959367/136152550-1ba265e2-f9c3-4dfe-ba31-7500388b826c.png)
Could you please suggest what I can do in order to create that shortcut for myself?

**Reply by abitofevrything** [reply]
Yes, that's my bad for saying it would be there. Some shortcuts are included in the CKEditor bundle (the editing component) and as such are not editable from the edit tab, but they are still available.

The shortcut to pull up the edit math dialog is Crtl+M.

**Reply by protean-onion** [reply]
I had to make a few adjustments to my system, but it worked! Thanks for your time. I appreciate it very much.
I am still interested in learning if I can automate tasks and create keyboard shortcuts for them. An example would be a set of keyboard shortcuts to highlight my text with different colors; maybe Ctrl+H+B for blue and Ctrl+H+O for orange. Is that possible?

**Reply by abitofevrything** [reply]
Interacting with the CKEditor component from the outside is not the easiest as it is minified and there are no utility functions in the API to access it.
It's not impossible though, so there should be a way to accomplish what you want.
